File: features/feed/feed/Feed.go
package feed

import (
	"github.com/hailongz/golang/db"
)

type Feed struct {
	db.Object
	Uid	int64	`json:"uid" name:"uid" title:"发布者ID"`
	Body	string	`json:"body" name:"body" title:"内容" length:"-1"`
	Options	interface{}	`json:"options,omitempty" name:"options" title:"其他数据" length:"-1"`
	Status	int32	`json:"status" name:"status" title:"状态" index:"ASC"`
	Ctime	int64	`json:"ctime" name:"ctime" title:"创建时间"`
	State	int32	`json:"state" name:"state" title:"回收状态" index:"ASC"`
}

func (O *Feed) GetName() string {
	return "feed"
}

func (O *Feed) GetTitle() string {
	return "动态"
}

File: features/feed/ts/Feed.ts
import { int64 } from "./lib/less";

export enum FeedStatus {
    
    /**
     * 新动态
     */
    None = 0,
    
    /**
     * 已分发完成
     */
    Completed = 1,
}

export enum FeedState {
    
    /**
     * 可用
     */
    None = 0,
    
    /**
     * 已回收
     */
    Recycle = 1
}

/**
 * 动态
 * @type db
 */
export class Feed {

    /**
     * ID
     */
    id: int64 = 0

    /**
     * 发布者ID
     * @index
     */
    uid: int64 = 0

    /**
     * 内容
     * @length -1
     */
    body: string = ""

    /**
     * 其他数据
     * @length -1
     */
    options: any

    /**
     * 状态
     * @index ASC
     */
    status: FeedStatus = FeedStatus.None

    /**
     * 创建时间
     */
    ctime: int64 = 0

    /**
     * 回收状态
     * @index ASC
     */
    state: FeedState = FeedState.None
    
}